### PR TITLE
Adds special click actions for Toggle Suit Sensors and Set Transfer Amount

### PR DESCRIPTION
**Pull request by CinnamonSnowball**
## What Does This PR Do
Alt+Shift+Clicking jumpsuits now allows you to toggle their suit sensors.
Alt+Clicking reagent containers now allows you to change their transfer amount.

Both had some minor adjacency sanity checks added to ensure you can't interact with them from far away/through solid border objects like directional windows.

## Why It's Good For The Game
Currently, the way to do both of these things is via the RightClick context menu, or through selecting their respective Object verbs often placed among a bunch of others. This is not very discoverable nor convenient, which isn't good for these fairly commonly used functions - especially now that since #18374, setting macros for these two verbs to alleviate the convenience issue is no longer possible.
Moving away from triggering special item functionality via the context menu to more responsive Alt(+Shift)+Click actions should be a Quality of Life tweak that also helps new players more easily discover how to do something as basic as making sure the station Paramedic knows when to save their ass.

## Changelog
:cl:
add: You can now Alt+Shift+Click jumpsuits to toggle their suit sensors
add: You can now Alt+Click beakers and other reagent containers to set their transfer amount
fix: You can no longer set reagent container transfer amounts remotely/toggle suit sensors through solid border objects
fix: Jumpsuits with no suit sensors no longer report disabled suit sensors on examine
/:cl:

### CODE REVIEW

File: code/modules/clothing/clothing.dm [hunk 1]
```diff
@@ -353,6 +353,10 @@ BLIND     // can't see anything
 	set src in usr
 	set_sensors(usr)
 
+/obj/item/clothing/under/AltClick(mob/user)
+	if(user.Adjacent(src))
+		set_sensors(user)
```
Comment (Contrabang) [inline code/modules/clothing/clothing.dm:358]: I would consider moving the `if(user.Adjacent(src))` check inside the set_sensors proc

File: code/modules/reagents/reagent_containers.dm [hunk 2]
```diff
@@ -25,9 +25,16 @@
...
 	var/N = input("Amount per transfer from this:", "[src]", default) as null|anything in possible_transfer_amounts
+	if(!usr.Adjacent(src))
+		to_chat(usr, "You have moved too far away.")
+		return
 	if(N)
 		amount_per_transfer_from_this = N
 
+/obj/item/reagent_containers/AltClick(mob/user)
+	if(user.Adjacent(src))
+		set_APTFT()
```
Comment (Contrabang) [inline code/modules/reagents/reagent_containers.dm:36]: ditto here

**Comment by Inadvizable** [comment]
I think alt+click should stay for accessories, and the more complex one should be suit sensors. You toggle your suit sensors usually once per round. You might be removing accessories (think badge, holster) many times.

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (hal9000PR) [reply code/modules/clothing/clothing.dm:358]: Don't do this. You can move while the input box is up, so two checks is needed.

**Comment by CinnamonSnowball** [comment]
> I think alt+click should stay for accessories, and the more complex one should be suit sensors. You toggle your suit sensors usually once per round. You might be removing accessories (think badge, holster) many times.

That _is_ something I thought about, but ultimately decided removing accessories is less mechanically important than keeping your sensors up (which is even recommended/required by the SOP). Can still revert that if the consensus is to keep old behavior.

File: code/modules/clothing/clothing.dm (hunk 1 above)
Comment (Contrabang) [reply code/modules/clothing/clothing.dm:358]: Yeah, I didnt mean to remove it, just to move it to the beginning of the set_sensors proc, but thats just a suggestion

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/modules/clothing/clothing.dm [hunk 3]
```diff
@@ -751,18 +754,26 @@ BLIND     // can't see anything
...
-		if(1)
-			. += "Its binary life sensors appear to be enabled."
-		if(2)
-			. += "Its vital tracker appears to be enabled."
-		if(3)
-			. += "Its vital tracker and tracking beacon appear to be enabled."
+
+	if(has_sensor >= 1)
+		switch(sensor_mode)
+			if(0)
```
Comment (farie82) [inline code/modules/clothing/clothing.dm:760]: Can you use the sensor defines here? Like `SUIT_SENSOR_OFF`

File: code/modules/clothing/clothing.dm [hunk 4]
```diff
@@ -308,7 +307,7 @@ BLIND     // can't see anything
 
 	var/list/modes = list("Off", "Binary sensors", "Vitals tracker", "Tracking beacon")
 	var/switchMode = input("Select a sensor mode:", "Suit Sensor Mode", modes[sensor_mode + 1]) in modes
-	if(get_dist(user, src) > 1)
+	if(!user.Adjacent(src))
```
Comment (farie82) [inline code/modules/clothing/clothing.dm:310]: Same here for the hands blocked thing

**Review by farie82 — CHANGES_REQUESTED** [review]

**Comment by matttheficus** [comment]
PR title could use a swap if the suit sensors change was reverted.

**Review by farie82 — APPROVED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**